### PR TITLE
perf: use custom HTTP Cache

### PR DESCRIPTION
**Pull request by ranjitmahadik**
**Summary:**  
PR contains a benchmark for execute method present in `http.rs`.
currently the benchmark only tests execute method with http_cache set to true.

**bench results:**
**before**: 
[38.054 µs 38.204 µs 38.428 µs]
[27.049 µs 27.363 µs 27.774 µs]
[26.795 µs 27.089 µs 27.492 µs]

**after**: 
[18.419 µs 18.432 µs 18.449 µs]
[17.414 µs 17.743 µs 18.018 µs]
[14.797 µs 14.876 µs 14.975 µs]

performance improvement : 51%.

**Issue Reference(s):**  
related to #1852

**Build & Testing:**

- [x] I ran `cargo test` successfully.
- [ ] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [x] I have added relevant unit & integration tests.
- [x] I have updated the [documentation] accordingly.
- [x] I have performed a self-review of my code.
- [x] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -35,3 +35,6 @@ pub use core::tracing::{
 };
 pub use core::valid::{Cause, ValidationError, Validator};
 pub use core::{Cache, EntityCache, EnvIO, FileIO, HttpIO, WorkerIO};
+
+#[cfg(feature = "cli")]
+pub use cli::runtime::NativeHttp;
```
Comment (tusharmath) [inline src/lib.rs:40]: We can't be using this feature flag.

Comment (tusharmath) [reply src/lib.rs:40]: Will use it for now. This is too much effort fixing.

File: Cargo.toml [hunk 2]
```diff
@@ -49,6 +49,8 @@ opentelemetry-otlp = { version = "0.15.0", features = [
     "tls-roots",
 ], optional = true }
 opentelemetry-system-metrics = { version = "0.1.8", optional = true }
+tailcall-http-cache = {path = "tailcall-http-cache", optional = true }
```
Comment (tusharmath) [inline Cargo.toml:52]:
```suggestion
tailcall-http-cache = {path = "tailcall-http-cache" }
```

Why is this optional?

File: tailcall-http-cache/src/http_cache_manager.rs [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+use std::sync::Arc;
+
+pub struct HttpCacheManager {
+    pub cache: Arc<MokaCache<String, Store>>,
+}
+
+impl Default for HttpCacheManager {
+    fn default() -> Self {
+        Self::new(MokaCache::new(42))
```
Comment (tusharmath) [inline tailcall-http-cache/src/http_cache_manager.rs:15]: Can you create an issue to configure the size of the HTTP cache via an upstream setting.

Comment (tusharmath) [reply tailcall-http-cache/src/http_cache_manager.rs:15]: before
```graphql
@upstream(httpCache: true)
```

after
```graphql
@upstream(httpCache: 128)
```

File: tailcall-http-cache/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,3 @@
+mod http_cache_manager;
+
+pub use http_cache_manager::HttpCacheManager;
```
Comment (tusharmath) [inline tailcall-http-cache/src/lib.rs:3]:
```suggestion
pub use cache::HttpCacheManager;
```

Can keep the name simple

File: Cargo.toml (hunk 2 above)
Comment (ranjitmahadik) [reply Cargo.toml:52]: as per comments in cargo.toml file, `http-cache-reqwest` has some conflicts with wasm and it's conditionally enabled for cli module and not keeping it optional was causing this integration to fail initially.

File: tailcall-http-cache/src/http_cache_manager.rs (hunk 3 above)
Comment (ranjitmahadik) [reply tailcall-http-cache/src/http_cache_manager.rs:15]: sure, will create an issue.

File: tailcall-http-cache/src/lib.rs (hunk 4 above)
Comment (ranjitmahadik) [reply tailcall-http-cache/src/lib.rs:3]: done, renamed module to cache.

**Review by tusharmath — APPROVED** [review]

**Pull request merged**